### PR TITLE
[dx12] get_fence_status

### LINKED ISSUE #2581: [dx12] Implement get_fence_status
This is currently a blocker for wgpu-rs on DX12

### PR DESCRIPTION
**Pull request by kvark**
Fixes #2581
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: src/backend/dx12/src/device.rs [hunk 1]
```diff
@@ -2867,16 +2867,14 @@ impl d::Device<B> for Device {
         }
     }
 
-    unsafe fn get_fence_status(&self, _fence: &r::Fence) -> Result<bool, d::DeviceLost> {
-        unimplemented!()
+    unsafe fn get_fence_status(&self, fence: &r::Fence) -> Result<bool, d::DeviceLost> {
+        Ok(fence.borrow().raw.GetCompletedValue() != 0)
```
Comment (msiglreith) [inline src/backend/dx12/src/device.rs]: why `borrow()`?

Comment (msiglreith) [reply src/backend/dx12/src/device.rs]: Would you mind adding the additional check against `UINT64_MAX` for device removal?

**Comment by kvark** [comment]
Thanks @msiglreith ! Please take another look.

**Comment by kvark** [comment]
uh, this is annoying. Gotta improve that Windows configuration somehow
bors retry

**Comment by kvark** [comment]
bors retry

**Comment by kvark** [comment]
bors r=msiglreith

**Comment by msiglreith** [comment]
meh timeouts randomly /:

**Pull request merged**